### PR TITLE
HPCC-18042 Unicode Implementation for ExcludeFirstWord

### PR DESCRIPTION
**Pull request by dskaff**
- Adds test cases to ecllibrary/teststd/uni/TestExcludeFirstWord.ecl
- Adds documentation to ecllibrary/std/Uni.ecl
- Adds code for ExcludeFirstWord to unicodelib.cpp

Signed-off-by: David Skaff <david.skaff@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by dskaff** [comment]
@johnholt please code review.

**Review by johnholt — APPROVED** [review]
OK

File: plugins/unicodelib/unicodelib.cpp [hunk 1]
```diff
@@ -83,6 +83,7 @@ static const char * EclDefinition =
 "  unsigned4 UnicodeLocaleWordCount(const unicode text, const varstring localename) : c, pure,entrypoint='ulUnicodeLocaleWordCount', hole; \n"
 "  unicode UnicodeLocaleGetNthWord(const unicode text, unsigned4 n, const varstring localename) : c,pure,entrypoint='ulUnicodeLocaleGetNthWord';\n"
 "  unicode UnicodeLocaleExcludeNthWord(const unicode text, unsigned4 n, const varstring localename) :c,pure,entrypoint='ulUnicodeLocaleExcludeNthWord';\n"
+"  unicode UnicodeLocaleExcludeFirstWord(const unicode text, const varstring localename) :c,pure,entrypoint='ulUnicodeLocaleExcludeFirstWord';\n"
```
Comment (ghalliday) [inline plugins/unicodelib/unicodelib.cpp]: I don't think this function is needed it isn't called by your code

File: plugins/unicodelib/unicodelib.cpp [hunk 2]
```diff
@@ -1475,4 +1476,25 @@ UNICODELIB_API void UNICODELIB_CALL ulUnicodeLocaleExcludeNthWord(unsigned & tgt
         tgtLen = 0;
         tgt = 0;
     }
+}
+
+UNICODELIB_API void UNICODELIB_CALL ulUnicodeLocaleExcludeFirstWord(unsigned & tgtLen, UChar * & tgt, unsigned textLen, UChar const * text, char const * localename)
```
Comment (ghalliday) [inline plugins/unicodelib/unicodelib.cpp]: this isn't called.

**Review by ghalliday — COMMENTED** [review]
@dskaff the unused function should be removed.

**Comment by ghalliday** [comment]
Also checkboxes are not completed on the PR.

**Comment by dskaff** [comment]
@ghalliday please code review

**Comment by ghalliday** [comment]
@dskaff please rebase onto latest master so this can be merged.

**Comment by dskaff** [comment]
@ghalliday should be ready for merge.

**Pull request merged**